yb-voyager/src/srcdb/postgres.go

```diff
@@ -1434,43 +1460,61 @@ func (pg *PostgreSQL) listSequencesMissingSelectPermission() (sequencesWithMissi
 	return sequencesWithMissingPerm, nil
 }
 
-func (pg *PostgreSQL) listTablesMissingSelectPermission() (tablesWithMissingPerm []string, err error) {
+func (pg *PostgreSQL) listTablesMissingSelectPermission(queryTableList string) (tablesWithMissingPerm []string, err error) {
 	// Users only need SELECT permissions on the tables of the schema they want to export for export schema
-	trimmedSchemaList := pg.getTrimmedSchemaList()
-	trimmedSchemaList = append(trimmedSchemaList, "pg_catalog", "information_schema")
-	querySchemaList := "'" + strings.Join(trimmedSchemaList, "','") + "'"
+	checkTableSelectPermissionQuery := ""
+	if queryTableList == "" {
+
+		trimmedSchemaList := pg.getTrimmedSchemaList()
+		trimmedSchemaList = append(trimmedSchemaList, "pg_catalog", "information_schema")
+		querySchemaList := "'" + strings.Join(trimmedSchemaList, "','") + "'"
+
+		checkTableSelectPermissionQuery = fmt.Sprintf(`
+		WITH schema_list AS (
+			SELECT unnest(ARRAY[%s]) AS schema_name
+		),
+		accessible_schemas AS (
+			SELECT schema_name
+			FROM schema_list
+			WHERE has_schema_privilege('%s', quote_ident(schema_name), 'USAGE')
+		)
+		SELECT
+			t.schemaname AS schema_name,
+			t.tablename AS table_name,
+			CASE 
+				WHEN has_table_privilege('%s', quote_ident(t.schemaname) || '.' || quote_ident(t.tablename), 'SELECT') 
+				THEN '%s' 
+				ELSE '%s' 
+			END AS status
+		FROM pg_tables t
+		JOIN accessible_schemas a ON t.schemaname = a.schema_name
+		UNION ALL
+		SELECT
+			t.schemaname AS schema_name,
+			t.tablename AS table_name,
+			'%s' AS status
+		FROM pg_tables t
+		WHERE t.schemaname IN (SELECT schema_name FROM schema_list)
+		AND NOT EXISTS (
+			SELECT 1
+			FROM accessible_schemas a
+			WHERE t.schemaname = a.schema_name
+		);`, querySchemaList, pg.source.User, pg.source.User, GRANTED, MISSING, NO_USAGE_PERMISSION)
+	} else {
+		checkTableSelectPermissionQuery = fmt.Sprintf(`
+		SELECT
+            quote_ident(split_part(t.schemaname || '.' || t.tablename, '.', 1)) AS schema_name,
```

Comment:
why can't we use `t.schemaname` and `t.tablename` directly ?

Reply:
Yeah I kept it this way since we are fetching the schema name and table name separately in the output of the query.